work on point crud operation and store on firebase and write testcases for crud operations

anshu-dev opened the pull request.

anshu-dev committed:
work on point crud operation and store on firebase and write testcases for crud operations

anshu-dev committed:
Add README.md

kasunvp commented on spec/requests/points_spec.rb:
```diff
@@ -0,0 +1,82 @@
+      expect(Point.all.size).to eq 0
+    end
+
+  end
```
:+1: on adding tests

kasunvp replied:
And using RSpec :100:

kasunvp commented:
@anshu-dev overall is good.
But you have missed some points from the requirements.

As per the requirements,

- We need to store the point in the encoded format
- We need the firestore database not the firebase realtime database
- The point doesn't get deleted from firebase

Also
- Use return status codes in the Rails way 
- Let's use instance variables where necessary

And let's add a bit more interesting functionality here.
- Add user authentication (register, login, password reset) using Doorkeeper, Devise and ember-simple-auth (on the ember end) 
- For password reset, the user should be given a url and by clicking that, the user will be able to reset the password.
- Add `created_by` relationship to the `Point`

More info on the web PR

kasunvp commented:
Another point I missed is that let's read constants from the ENV, eg: the URL you have for Firebase.
Same thing is valid for Ember

anshu-dev committed:
WIP PR requested changes
Added devise, doorkeeper, dotenv-rails gem
Used rails way status code
Added user model

anshu-dev committed:
Added current_user and search action

kasunvp commented:
@anshu-dev also note that point is a single geometry we need to store that as it is not as separate longitude and latitude

anshu-dev committed:
Moved PointsController inside api folder

anshu-dev committed:
Added user_params in registrationscontroller

anshu-dev committed:
Moved points routes in api scope

anshu-dev committed:
Added concern json_api_deserialization

anshu-dev committed:
Added user_params in sessionscontroller

anshu-dev committed:
Updated sessions controller

anshu-dev committed:
Added session create action

anshu-dev committed:
Used find_for_database_authentication

anshu-dev committed:
Added login functionality

anshu-dev committed:
Added password reset functionality and Geometry to store longitude and latitude

anshu-dev committed:
Modified index action in pointscontroller for searching

anshu-dev committed:
Added smtp configuration in production.rb

anshu-dev committed:
Setup firestore

anshu-dev committed:
Fixed typo

anshu-dev committed:
Code clean-up

anshu-dev committed:
Fixed rspecs

anshu-dev committed:
Password edit url changes

anshu-dev committed:
Removed unnecessary code
